### PR TITLE
👌 Shift time axis by IRF location

### PR DESCRIPTION
**Pull request by s-weigand**
Since the time axis of the concentration plot is shifted so the location of the IRF is at 0 in overview plots, this makes it harder to compare it the residual/data and LSV plots, especially when using `linlog` axes.
This PR adds the argument `irf_location` to the low level plotting functions which shifts the time axis for those plots the same way the time axis of concentrations is shifted. This shift is automatically applied in `plot_overview` and `plot_simple_overview`.
It also allows for shifting the time axis for the `plot_data_overview` function, which will make it easier to "guesstimate" the initial center location of the IRF, especially when used together with `linlog` scaling.

### Before
image
image

### After
image
image
image

### Change summary

- [♻️ Factored out time shifting by irf location into seperate function](https://github.com/glotaran/pyglotaran-extras/commit/3649fc8014e952f4e5db4721d780040fae6a92fc)
- [👌 Added time shifting by irf location tp plot functions](https://github.com/glotaran/pyglotaran-extras/commit/35debe530c619a8929bcc465fa6dcb8b029b9c2c)
- [🩹 Apply linlog scaling to LSV plot in 'plot_data_overview'](https://github.com/glotaran/pyglotaran-extras/pull/99/commits/2c01e723ba9f15bacf5a77946120282381aceb14)

### Checklist

- [x] ✔️ Passing the tests (mandatory for all PR's)

### CODE REVIEW

**Comment by s-weigand** [comment]
I used [BE-case study](https://github.com/ism200/PE2022/tree/main/BE-case-study) for the example plots.
And tests run against the PE2022 teaching material can be found [here](https://github.com/s-weigand/PE2022/actions/runs/2606308444)

**Comment by jsnel** [comment]
Nice work. 

Just one question, is it possible to customize the color and stroke thickness of the time zero line that visualizes the dispersion of the IRF? I can imagine people wishing a more subtle line, or one that doesn't clash with the data itself.

**Comment by s-weigand** [comment]
> Just one question, is it possible to customize the color and stroke thickness of the time zero line that visualizes the dispersion of the IRF? I can imagine people wishing a more subtle line, or one that doesn't clash with the data itself.

Currently not, but this was part of #95 , not this PR.
Maybe in a flow up PR we can add a `irf_center_plot_style` argument

**Comment by s-weigand** [comment]
[Pre-commit CI doesn't like](https://results.pre-commit.ci/run/github/299106891/1656881091.C3qto1FIQHKGuVUPU-cQvw) how long the doc linting takes 😅🙈

File: pyglotaran_extras/plotting/utils.py [hunk 1]
```diff
@@ -287,6 +287,43 @@ def extract_dataset_scale(res: xr.Dataset, divide_by_scale: bool = True) -> floa
...
+    irf_location: float | None,
+) -> xr.DataArray:
+    """Shift ``plot_data`` 'time' axis  by the position of the main ``irf``.
+
+    Parameters
+    ----------
+    plot_data: xr.DataArray
+        Data to plot.
+    irf_location:  float | None
+        Location of the ``irf``, if the value is None the original ``plot_data`` will be returned.
```
Comment (jsnel) [inline pyglotaran_extras/plotting/utils.py:301]: We could consider being more verbose, writing "Center location of the ``irf```, ... " 
The issue is that the 'location' of an 'IRF' is hard to define, considering the multi-dimensionality of it, but that's probably something that should be documented in pyglotaran, not in the -extras. 

To be clear, we generally mean:

The center of the main Gaussian band that described the majority of the amplitude of the instrument response function, as some (user-defined) global coordinate (e.g. center wavelength / center lambda).

**Review by jsnel — APPROVED** [review]
Looks good. Nice improvement to the plots.

**Pull request merged**